purchases/src/main/kotlin/com/revenuecat/purchases/DangerousSetting.kt

```diff
@@ -0,0 +1,5 @@
+package com.revenuecat.purchases
+
+enum class DangerousSetting {
```

Comment:
Is this a good idea?

Reply:
I kind of like this 😊

Reply:
yes? Alternative was `InternalSetting`. What do you think?

Reply:
I prefer `Dangerous`, I think it makes it abundantly clear that you're not supposed to mess with it. 
But I do think that we should add a note here telling folks to NOT use them unless they absolutely know what they're doing and have contacted RC support team first

Reply:
Good idea about the note

Reply:
I tried adding a build warning but the only feasible way of doing it in Java world is deprecating it, which I think it's not something we want to do. We could also annotate it with `VisibleForTesting`

Reply:
maybe we could specify that this is auto sync specifically for purchases? like, subs attributes and other stuff will continue to work

Reply:
i.e.: `AUTO_SYNC_PURCHASES_OFF`